autoextract_poet/page_inputs.py

```diff
@@ -46,12 +46,25 @@ class AutoExtractData(Generic[T]):
 
     @property
     def item_class(self):
-        return self.__orig_bases__[0].__args__[0]
+        return get_item_class(self)
 
     def to_item(self) -> Optional[T]:
         return self.item_class.from_dict(self.data[self.page_type])
 
 
+def get_item_class(page_input_cls):
```

Comment:
What do you think about adding type annotations here?
```suggestion
def get_item_class(page_input_cls: Type[AutoExtractData]) -> Type[Item]:
```